docs/protocol-basics.rst

```diff
@@ -138,8 +136,60 @@ block public key.
 
 .. _status:
 
-Status
-------
+Status (protocol 1.5 and later)
+-------------------------------
+
+To calculate the `status` of a :ref:`script hash <script hashes>` (or
+address):
+
+1. Consider all transactions touching the script hash (both those spending
+from it, and those funding it), both confirmed and unconfirmed (in mempool).
+
+2. Order confirmed transactions by increasing height (and position in the
+block if there are more than one in a block).
+
+3. For mempool transactions, we define **height** to be ``-1`` if the
+transaction has at least one unconfirmed input, and ``0`` if all inputs are
+confirmed.
+
+4. Order mempool transactions by ``(-height, tx_hash)``, that is,
+``0`` height txs come before ``-1`` height txs, and secondarily the
+txid (in network byteorder) is used to arrive at a canonical ordering.
+
+5. For each confirmed tx, form a bytearray: ``tx_hash+height``, where:
+
+  * ``tx_hash`` is the 32-byte transaction hash in network byteorder
+    (reverse of human display endianness)
+
+  * ``height`` is the height of the block the tx is included in,
+    serialised as 8-bytes, little-endian, signed (two's complement)
+
+6. For each mempool tx, form a bytearray: ``tx_hash+height+fee``, where:
+
+  * ``tx_hash`` is the 32-byte transaction hash in network byteorder
+    (reverse of human display endianness)
+
+  * ``height`` is either ``0`` or ``-1``, as defined above,
+    serialised as 8-bytes, little-endian, signed (two's complement)
+
+  * ``fee`` is the transaction fee in minimum coin units (satoshis),
+    serialised as 8-bytes, little-endian, unsigned
+
+7. The :dfn:`status` of the script hash is defined by the following recursion:
+
+  * ``status_0`` is 32 zero bytes
+
+  * ``status_n`` is calculated as ``sha256(status_(n-1) + tx_n)``
+
+  That is, for a script hash with an empty history, the status is ``status_0``.
+  If the history contains ``n`` txs, the status is ``status_n``. The ``tx_n``
+  series consists of, first the confirmed txs touching the script hash,
+  followed by the mempool txs touching it; formatted as described above, as
+  bytearrays of length 40 or 48.
```

Comment:
I see long-term problems mixing mempool and confirmed history here; I would recommend moving to separate tracking of mempool and confirmed histories.  Of course that is more complexity in the short-term.

Reply:
Do you mean specifically for the status, for `get_history`, or in general?

Reply:
In general